implementation/src/main/java/io/smallrye/mutiny/helpers/spies/MultiGlobalSpy.java

```diff
@@ -0,0 +1,95 @@
+package io.smallrye.mutiny.helpers.spies;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.subscription.MultiSubscriber;
+
+public class MultiGlobalSpy<T> extends MultiSpyBase<T> {
+
+    private final MultiOnCancellationSpy<T> onCancellationSpy;
+    private final MultiOnCompletionSpy<T> onCompletionSpy;
+    private final MultiOnFailureSpy<T> onFailureSpy;
+    private final MultiOnItemSpy<T> onItemSpy;
+    private final MultiOnRequestSpy<T> onRequestSpy;
+    private final MultiOnSubscribeSpy<T> onSubscribeSpy;
+    private final MultiOnTerminationSpy<T> onTerminationSpy;
+
+    MultiGlobalSpy(Multi<T> upstream) {
+        super(upstream);
+        onCancellationSpy = Spy.onCancellation(upstream);
+        onCompletionSpy = Spy.onCompletion(onCancellationSpy);
+        onFailureSpy = Spy.onFailure(onCompletionSpy);
+        onItemSpy = Spy.onItem(onFailureSpy);
+        onRequestSpy = Spy.onRequest(onItemSpy);
+        onSubscribeSpy = Spy.onSubscribe(onRequestSpy);
+        onTerminationSpy = Spy.onTermination(onSubscribeSpy);
+    }
+
+    public MultiOnCancellationSpy<T> onCancellationSpy() {
```

Comment:
Wondering if the `spy` suffix is necessary.

Reply:
Yes, because the spy is also a `Multi` (or a `Uni`) so method names would clash.